sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/AppendBlobAPITest.groovy

```diff
@@ -578,6 +580,40 @@ class AppendBlobAPITest extends APISpec {
         null                  | null                    | null          | receivedEtag
     }
 
+    def "Append block from URL source oauth"() {
+        setup:
+        def sourceBlob = cc.getBlobClient(generateBlobName())
+        sourceBlob.upload(data.defaultBinaryData)
+        def oauthHeader = getAuthToken()
+
+        when:
+        bc.appendBlockFromUrlWithResponse(
+            new AppendBlobAppendBlockFromUrlOptions(sourceBlob.getBlobUrl())
+                .setSourceAuthorization(new HttpAuthorization("Bearer", oauthHeader)),
+            null, Context.NONE)
+
+        then:
+        def os = new ByteArrayOutputStream(data.defaultDataSize)
+        bc.download(os)
+        os.toByteArray() == data.defaultBytes
+    }
+
+    def "Append block from URL source oauth fail"() {
+        setup:
+        def sourceBlob = cc.getBlobClient(generateBlobName())
+        sourceBlob.upload(data.defaultBinaryData)
+        def oauthHeader = "garbage"
+
+        when:
+        bc.appendBlockFromUrlWithResponse(
+            new AppendBlobAppendBlockFromUrlOptions(sourceBlob.getBlobUrl())
+                .setSourceAuthorization(new HttpAuthorization("Bearer", oauthHeader)),
+            null, Context.NONE)
+
+        then:
+        thrown(BlobStorageException)
+    }
+
```

Comment:
Should these go to new class ?

Reply:
It looks like I pressed ctrl-c instead of ctrl-x. Will delete. I have explicitly deleted the recordings for these tests already.